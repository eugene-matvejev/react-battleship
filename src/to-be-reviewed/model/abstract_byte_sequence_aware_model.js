export default class AbstractByteSequenceAwareModel {
    constructor() {
        this.byteSeq = 0x00;
    }

    getSequence() {
        return this.byteSeq;
    }

    setSequence(seq) {
        this.byteSeq = seq;
    }

    addSequence(seq) {
        this.byteSeq |= seq;
    }

    removeSequence(seq) {
        this.byteSeq &= ~seq;
    }

    hasSequence(seq) {
        return (this.byteSeq & seq) === seq;
    }
}
