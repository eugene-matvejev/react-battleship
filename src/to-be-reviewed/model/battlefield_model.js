import AbstractByteSequenceAwareModel from './abstract_byte_sequence_aware_model';

export default class BattlefieldModel extends AbstractByteSequenceAwareModel {
    constructor() {
        super();

        this.player = undefined;
        this.id = undefined;
        this.size = 0;

        this.cellsIndexedById = {};
        this.cellsIndexedByCoordinate = {};
    }

    /**
     * @param {PlayerModel} player
     */
    setPlayer(player) {
        this.player = player;
    }

    /**
     * @return {PlayerModel}
     */
    getPlayer() {
        return this.player;
    }

    hasCellByCoordinate(v) {
        return undefined !== this.cellsIndexedByCoordinate[v];
    }

    hasCellById(v) {
        return undefined !== this.cellsIndexedById[v];
    }

    getCellByCoordinate(v) {
        return this.cellsIndexedByCoordinate[v];
    }

    getCellById(v) {
        return this.cellsIndexedById[v];
    }

    getCellsIndexedById() {
        return this.cellsIndexedById;
    }

    getCellsIndexedByCoordinate() {
        return this.cellsIndexedByCoordinate;
    }

    /**
     * @param {{id: number, coordinate: string}[]} cells
     */
    setCells(cells) {
        this.cellsIndexedByCoordinate = {};
        this.cellsIndexedById = {};

        for (const cell of cells) {
            this.cellsIndexedByCoordinate[cell.coordinate] = cell;

            if (undefined !== cell.id) {
                this.cellsIndexedById[cell.id] = cell;
            }
        }
    }
}
