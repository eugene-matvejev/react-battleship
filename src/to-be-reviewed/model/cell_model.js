import AbstractByteSequenceAwareModel from './abstract_byte_sequence_aware_model';

const FLAG_NONE = 0x00;
const FLAG_DEAD = 0x01;
const FLAG_SHIP = 0x02;
/** const FLAG_DEAD_SHIP = FLAG_SHIP | FLAG_DEAD; */
const FLAG_SKIP = 0x04;

export default class CellModel extends AbstractByteSequenceAwareModel {
    constructor(coordinate) {
        super();

        this.coordinate = coordinate;
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
    }

    getCoordinate() {
        return this.coordinate;
    }

    static getCoordinateCharacter = (coordinate) => coordinate.charAt(0);

    static getCoordinateDigit = (coordinate) => parseInt(coordinate.substring(1), 10)

    static flags = {
        none: FLAG_NONE,
        dead: FLAG_DEAD,
        ship: FLAG_SHIP,
        skip: FLAG_SKIP
    }
}
