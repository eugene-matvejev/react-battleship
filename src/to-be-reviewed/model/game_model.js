import AbstractByteSequenceAwareModel from './abstract_byte_sequence_aware_model';

export default class GameModel extends AbstractByteSequenceAwareModel {
    constructor() {
        super();

        this.battlefields = [];
    }

    addBattlefield(battlefield) {
        this.battlefields.push(battlefield);
    }

    getBattlefields() {
        return this.battlefields;
    }
}
