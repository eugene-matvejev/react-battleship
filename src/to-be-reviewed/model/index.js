import AbstractByteSequenceAwareModel from './battlefield_model';
import BattlefieldModel from './battlefield_model';
import CellModel from './cell_model';
import GameModel from './game_model';
import PlayerModel from './player_model';

export {
    AbstractByteSequenceAwareModel,
    BattlefieldModel,
    CellModel,
    GameModel,
    PlayerModel,
}
