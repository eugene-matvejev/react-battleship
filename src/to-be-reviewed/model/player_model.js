import AbstractByteSequenceAwareModel from './abstract_byte_sequence_aware_model';

const FLAG_HUMAN_CONTROLLED = 0x01;

export default class PlayerModel extends AbstractByteSequenceAwareModel {
    constructor() {
        super();

        this.id = 'undefined';
        this.name = 'undefined';
        this.score = 0;
        this.avatarSrc = `/assets/img/avatar-placeholder.png`;
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
    }

    getName() {
        return this.name;
    }

    setName(v) {
        this.name = v;
    }

    getScore() {
        return this.score;
    }

    setScore(v) {
        this.score = v;
    }

    getAvatarSrc() {
        return this.avatarSrc;
    }

    setAvatarSrc(v) {
        this.avatarSrc = v;
    }

    isHumanControlled() {
        return this.hasSequence(this.constructor.getHumanFlag());
    }

    static getHumanFlag = () => FLAG_HUMAN_CONTROLLED;
}
