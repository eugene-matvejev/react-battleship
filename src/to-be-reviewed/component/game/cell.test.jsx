import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Cell from './cell';
import { CellModel } from '../../model';

configure({ adapter: new Adapter() });

describe(`<Cell/>`, () => {
    const model = new CellModel('A1');

    describe(`rendering`, () => {
        it('with default/required props', () => {
            shallow(<Cell model={model} />);
        });

        it('match snapshot', () => {
            const c = shallow(<Cell model={model} />);

            expect(c).toMatchSnapshot();
        });
    });
});
