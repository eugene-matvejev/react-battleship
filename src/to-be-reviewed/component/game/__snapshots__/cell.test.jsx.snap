// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell/> rendering match snapshot 1`] = `
<div
  className="component cell "
  data-byte-sequence={0}
/>
`;
